tide-whisperer.go

```diff
@@ -208,14 +220,6 @@ func main() {
 			return
 		}
 
-		token := req.Header.Get("x-tidepool-session-token")
-		td := shorelineClient.CheckToken(token)
-
-		if td == nil || !(td.IsServer || td.UserID == queryParams.UserId || userCanViewData(td.UserID, queryParams.UserId)) {
```

Comment:
Wait, I think you lost the `userCanViewData(td.UserID, queryParams.UserId)` check. The `authorize` handler checks `td == nil || !(td.IsServer || td.UserID == queryParams.UserId)` but not the latter part. I think you've just eliminated custodians and shares from viewing the data.